moveit_planners/pilz_industrial_motion_planner/include/pilz_industrial_motion_planner/command_list_manager.h

```diff
@@ -98,9 +102,11 @@ class CommandListManager
    *
    * @return Contains the calculated/generated trajectories.
    */
-  RobotTrajCont solve(const planning_scene::PlanningSceneConstPtr& planning_scene,
-                      const planning_pipeline::PlanningPipelinePtr& planning_pipeline,
-                      const moveit_msgs::MotionSequenceRequest& req_list);
+  RobotTrajCont solve(
+      const planning_scene::PlanningSceneConstPtr& planning_scene,
+      const planning_pipeline::PlanningPipelinePtr& planning_pipeline,
+      const moveit_msgs::MotionSequenceRequest& req_list,
+      const ItemPlannedCallback& on_item_planned = [](const planning_interface::MotionPlanResponse& /*unused*/) {});
```

Comment:
Can't you pass an empty function object here, i.e. `ItemPlannedCallback()`?

Reply:
From std::function documentation:

 > If a std::function contains no target, it is called empty. Invoking the target of an empty std::function results in [std::bad_function_call](https://en.cppreference.com/w/cpp/utility/functional/bad_function_call) exception being thrown.

I wanted to make sure that the default would be a "do nothing" operation, rather than an error to preserve the behaviour of existing code.